Makefile

```diff
@@ -119,12 +122,34 @@ test-containerized: run-etcd run-k8s-apiserver build-containerized dist/host-loc
 	-e ETCD_IP=$(LOCAL_IP_ENV) \
 	-e LOCAL_USER_ID=0 \
 	-e PLUGIN=calico \
+	-e CNI_SPEC_VERSION=$(CNI_SPEC_VERSION) \
 	-v $(CURDIR):/go/src/github.com/projectcalico/cni-plugin:rw \
 	$(CALICO_BUILD) sh -c '\
 			cd  /go/src/github.com/projectcalico/cni-plugin && \
 			ginkgo'
 	make stop-etcd
 
+
+run-test-containerized-without-building: run-etcd run-k8s-apiserver
+	docker run --rm --privileged --net=host \
+	-e ETCD_IP=$(LOCAL_IP_ENV) \
+	-e LOCAL_USER_ID=0 \
+	-e PLUGIN=calico \
+	-e CNI_SPEC_VERSION=$(CNI_SPEC_VERSION) \
+	-v $(CURDIR):/go/src/github.com/projectcalico/cni-plugin:rw \
+	$(CALICO_BUILD) sh -c '\
+			cd  /go/src/github.com/projectcalico/cni-plugin && \
+			ginkgo'
+	make stop-etcd
+
+## Run the tests in a container (as root) for different CNI spec versions
+## to make sure we don't break backwards compatiblity.
+.PHONY: test-containerized-cni-versions
+test-containerized-cni-versions: build-containerized dist/host-local;
+	for cniversion in "0.2.0" "0.3.1" ; do \
```

Comment:
I personally think this is ok since 0.2.0 would be the default if it's not specified, and we're changing all our manifests to use 0.3.1